src/main/java/org/kiwiproject/dropwizard/util/health/ServicePingHealthCheck.java

```diff
@@ -0,0 +1,198 @@
+ */
+@Slf4j
+public class ServicePingHealthCheck extends HealthCheck {
+
+    /**
+     * Indicator of how important a dependent service is to the current service.
+     * <ul>
+     *     <li>REQUIRED indicates that the current service cannot function without the dependency.</li>
+     *     <li>
+     *         WANTED indicates that the current service wants the dependency to function, but can operate in degraded
+     *         mode without it.
+     *     </li>
+     *     <li>
+     *         INFORMATIONAL indicates that the current service uses the dependency for informational purposes and
+     *         not for critical logic.
+     *     </li>
+     * </ul>
+     */
+    public enum DependencyImportance {
+        INFORMATIONAL(HealthStatus.INFO),
+        WANTED(HealthStatus.WARN),
+        REQUIRED(HealthStatus.CRITICAL);
+
+        private final HealthStatus healthStatus;
+
+        DependencyImportance(HealthStatus healthStatus) {
+            this.healthStatus = healthStatus;
+        }
+    }
+
+    private final ServiceIdentifier serviceIdentifier;
+    private final RegistryAwareClient client;
+
+    @VisibleForTesting
+    final DependencyImportance dependencyImportance;
+
+    /**
+     * Creates the health check set up to ping a service defined by the given {@link ServiceIdentifier} using the given
+     * {@link RegistryAwareClient}. By default the service to be checked will be marked as REQUIRED.
+     *
+     * @param serviceIdentifier the identifier for the service to be pinged.
+     * @param client            the {@link RegistryAwareClient} to use for service lookup and to perform the request
+     */
+    public ServicePingHealthCheck(ServiceIdentifier serviceIdentifier, RegistryAwareClient client) {
+        this(serviceIdentifier, client, DependencyImportance.REQUIRED);
+    }
+
+    /**
+     * Creates the health check set up to ping a service defined by the given {@link ServiceIdentifier} using the given
+     * {@link RegistryAwareClient} and marking the importance of the dependency by the given
+     * {@link DependencyImportance}.
+     *
+     * @param serviceIdentifier     the identifier for the service to be pinged.
+     * @param client                the {@link RegistryAwareClient} to use for service lookup and to perform the request
+     * @param dependencyImportance  the importance of the dependent service to the current service
+     */
+    public ServicePingHealthCheck(ServiceIdentifier serviceIdentifier,
+                                  RegistryAwareClient client,
+                                  DependencyImportance dependencyImportance) {
+        this.serviceIdentifier = requireNotNull(serviceIdentifier, "serviceIdentifier is required");
+        this.client = requireNotNull(client, "client is required");
+        this.dependencyImportance = requireNotNull(dependencyImportance, "dependencyPriority is required");
```

Comment:
I suggest adding this as a health check detail, so that it's easy to tell which services have which importance